Lib/importlib/_bootstrap_external.py

```diff
@@ -702,6 +702,11 @@ def spec_from_file_location(name, location=None, *, loader=None,
                 pass
     else:
         location = _os.fspath(location)
+        if not _path_isabs(location):
```

Comment:
`_path_isabs` needs to return false for a rooted path that has no drive, which is relative to the drive of the current working directory. It might seem to work if one only tests a DLL/PYD that has no dependencies. But the loader will fail the call as soon as it has to compute the DLL search path to find dependencies.

In POSIX, `_path_isabs` just needs to check for an initial slash. In Windows, it needs to check for two initial slashes (UNC) or a drive ("X:") followed by a slash. This allows paths that aren't valid filesystem paths, but entries in `sys.path` are assumed to be valid.